src/utils/external-extensions.ts

```diff
@@ -11,16 +11,29 @@ export const getDataFromExternalExtensionArgument = (externalExtension: string)
     externalExtension = getArgumentFromExternalExtensionOption(CURATED_EXTENSIONS[externalExtension]);
   }
 
+  const isGithubUrl = externalExtension.startsWith("https://github.com/");
+
   // Check format: owner/project:branch (branch is optional)
   const regex = /^[^/]+\/[^/]+(:[^/]+)?$/;
-  if (!regex.test(externalExtension)) {
-    throw new Error(`Invalid extension format. Use "owner/project" or "owner/project:branch"`);
+  if (!regex.test(externalExtension) && !isGithubUrl) {
+    throw new Error(`Invalid extension format. Use "owner/project", "owner/project:branch" or github url.`);
   }
 
-  // Extract owner, project and branch
-  const owner = externalExtension.split("/")[0];
-  const project = externalExtension.split(":")[0].split("/")[1];
-  const branch = externalExtension.split(":")[1];
+  let owner;
+  let project;
+  let branch;
+
+  if (isGithubUrl) {
+    // Extract owner, project and branch from github url
+    owner = externalExtension.split("/")[3];
```

Comment:
nit1: probably its better to split `externalExtension` before if block so we don't need to split it every time

nit2: probably we need to move this functionality of getting owner/project/branch to the function so it will be used the same way everywhere. Because for example there it works a bit differently (yes, there's an URL without a branch, but we can consider that case too)

https://github.com/scaffold-eth/create-eth/blob/7e323702c60a602a3377ba9aec174d0b6e424284/src/utils/external-extensions.ts#L75-L77

Reply:
Thanks for the review!

nit2: It can be useful, or maybe it's too much since we only use it twice. Let's wait and see what others think.

Reply:
> nit2: probably we need to move this functionality of getting owner/project/branch to the function so it will be used the same way everywhere. Because for example there it works a bit differently (yes, there's an URL without a branch, but we can consider that case too)

Yup I think it makes sense, maybe something very simple like: 

```js
function deconstructGithubUrl(url) {
  const urlParts = url.split('/');
  const ownerName = urlParts[3];
  const repoName = urlParts[4];
  const branch = urlParts[5] === 'tree' ? urlParts[6] : undefined;

  return { ownerName, repoName, branch };
}
```

Since we know that that `ownerName`, `repoName` would be always present at 3, 4 instead of doing lenth-2 ... so that we keep the logic same?